src/lib.rs

```diff
@@ -63,12 +63,13 @@ fn find_repos(root: &Path) -> Result<Option<(Vec<repo::Repo>, Vec<String>, bool)
                             let name = remote.name();
                             let url = remote.url();
                             let remote_type = match repo::detect_remote_type(&url) {
-                                Some(t) => t,
-                                None => {
+                                Ok(t) => t,
+                                Err(e) => {
                                     warnings.push(format!(
-                                        "{}: Could not detect remote type of \"{}\"",
+                                        "{}: Could not detect remote type of \"{}\". Reason: {}",
```

Comment:
The wording "Could not detect remote type" is not (and never was) correct here. In case of a `http` URL for example, you'd get the following, confusing error:

> Could not detect remote type of "http://example.com". Reason: Remotes using HTTP protocol are not supported

I'm not certain about a better wording, but something generic like the following should be good enough:

```suggestion
                                        "{}: Could not handle URL: {}",
```